src/pages/ServiceList/ServiceListComponent.tsx

```diff
@@ -1,67 +1,267 @@
 import * as React from 'react';
-import { ListView, ListViewItem, ListViewIcon } from 'patternfly-react';
+import { ListView, ListViewItem, ListViewIcon, Sort } from 'patternfly-react';
 import { Link } from 'react-router-dom';
+import { ServiceFilter, ServiceFilterSelected } from '../../components/ServiceFilter/ServiceFilter';
+import { Paginator } from 'patternfly-react';
+import { ActiveFilter } from '../../types/ServiceFilter';
+import * as API from '../../services/Api';
+import { Namespace } from '../../types/Namespace';
+import { Pagination, ServiceItem, ServiceList } from '../../types/ServiceListComponent';
+import PropTypes from 'prop-types';
+
+type SortField = {
+  id: string;
+  title: string;
+  isNumeric: boolean;
+};
+
+const sortFields: SortField[] = [
+  {
+    id: 'namespace',
+    title: 'Namespace',
+    isNumeric: false
+  },
+  {
+    id: 'servicename',
+    title: 'Service Name',
+    isNumeric: false
+  }
+];
+
+type ServiceListComponentState = {
+  loading: boolean;
+  services: ServiceItem[];
+  pagination: Pagination;
+  currentSortField: SortField;
+  isSortAscending: boolean;
+};
+
+type ServiceListComponentProps = {
+  onError: PropTypes.func;
+};
+
+class ServiceListComponent extends React.Component<ServiceListComponentProps, ServiceListComponentState> {
+  constructor(props: ServiceListComponentProps) {
+    super(props);
+    this.filterChange = this.filterChange.bind(this);
+    this.handleError = this.handleError.bind(this);
+    this.pageSet = this.pageSet.bind(this);
+    this.pageSelect = this.pageSelect.bind(this);
+    this.updateSortField = this.updateSortField.bind(this);
+    this.updateSortDirection = this.updateSortDirection.bind(this);
+    this.state = {
+      loading: true,
```

Comment:
In my experience it's better to set the `loading: true ` flag not in constructor, but at the beginning of the function that actually loads stuff. For instance here, loading is triggered both via `componentWillMount` and `filterChange`. I bet the latter won't show the spinner because of that.

Reply:
Mmm locally is working, but I will take a look.

Reply:
Ok, true, fix it.

Reply:
I updated in componentWillMount and filterChange method as well.
Just leave here a default value in the constructor.